### PR TITLE
add new default build configurations and fix build breaks

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. external/libtuv, mosquitto: use set_errno instead of errno 
2. configs/imxrt1020-evk/SAM: add new default build configurations

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [89fde066f25fc75e501418b3c56124526eb013ba] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [89fde066f25fc75e501418b3c56124526eb013ba] - Code Rule Check (C++) OK.

File: external/mosquitto/net_mosq.c [hunk 1]
```diff
@@ -1192,7 +1192,7 @@ int _mosquitto_packet_read(struct mosquitto *mosq)
 			mosq->in_packet.pos += read_length;
 		} else {
 #ifdef WIN32
-			errno = WSAGetLastError();
+			set_errno(WSAGetLastError();
```
Comment (NilabhShant) [inline external/mosquitto/net_mosq.c]: This statement requires one more close brace at the end.

**Comment by sunghan-chang** [comment]
@NilabhShant You're right. I fix them. Thank your for comments.

**Comment by seinfra** [comment]
Target : [b7f7c0a659f86bb3e4f8da1c3f020cefddeee233] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b7f7c0a659f86bb3e4f8da1c3f020cefddeee233] - Code Rule Check OK.

**Pull request merged**